crates/bevy_ui/src/render/mod.rs

```diff
@@ -289,11 +316,8 @@ pub fn extract_text_uinodes(
     let scale_factor = windows.scale_factor(WindowId::primary()) as f32;
 
     for (entity, uinode, transform, text, visibility, clip) in uinode_query.iter() {
-        if !visibility.is_visible {
-            continue;
-        }
-        // Skip if size is set to zero (e.g. when a parent is set to `Display::None`)
-        if uinode.size == Vec2::ZERO {
+        // Skip if the node is invisible or if its size is set to zero (e.g. when a parent is set to `Display::None`)
+        if !visibility.is_visible || uinode.size == Vec2::ZERO {
```

Comment:
This change could also be split out.

Reply:
I'll create a separate PR for this kind of improvements